htlcswitch/switch.go

```diff
@@ -351,25 +357,23 @@ type resolutionMsg struct {
 // didn't need to go to the chain in order to fulfill a contract. We'll process
 // this message just as if it came from an active outgoing channel.
 func (s *Switch) ProcessContractResolution(msg contractcourt.ResolutionMsg) error {
-
-	done := make(chan struct{})
+	errChan := make(chan error)
```

Comment:
If the switch quits here then an error is returned and the contractcourt won't get out of `StateContractClosed` and will re-send the next time (on startup)

Reply:
I don't understand - can you elaborate?

Reply:
Even if multiple selects are ready, that shouldn't matter since the err channel is used, which is signalled iff the save was successful or error if failed

Reply:
Maybe I'm overthinking here, what happens if multiple selects are ready + lnd going down?